proxy_server/authentication/auth.py

```diff
@@ -79,7 +79,16 @@ def get_and_delete_messages(self):
         return list()
 
     def has_perm(self, perm, obj=None):
-        return perm in self.roles
+        for permission in self.permissions:
+            if permission == perm:
+                return True
+        return False
+    def has_perms(self, perm_list, obj=None):
+        for perm in perm_list:
+            for permission in self.permissions:
```

Comment:
Esto también puede quedar así:

for perm in perm_list:
    if perm in self.permissions:
        return True
return False